### PR TITLE
Arcconf Plugin patch: Fix for 'volumes' function

### PR DESCRIPTION
**Pull request by sharath-t**
issue: column 'size' is showing wrong value
		fix: changed num_of_blocks calculation equation from [(dataSpace*1024)/blockSize] to dataSpace(or chunk->0->numBlocks)
		
issue: pool ID is incorrect for the LDs which are part of the same array
		fix: read 'consumedArrayId' in LD->chunk->0 to get array id
		
issue: column 'id' is not unique. volumes created out of different controllers has repeated IDs
		fix: volume ID is prefixed with system ID [systemID:volumeID]

### CODE REVIEW

File: plugin/arcconf/arcconf.py [hunk 1]
```diff
@@ -468,14 +471,15 @@ def volumes(self, search_key=None, search_value=None,
...
-                                                       sys_id, cnt,
-                                                       str(ld_num),
-                                                       ld_name)
+                    raid_level = ld_info['raidLevel']
+                    chunk_data = ld_info['Chunk']
+                    for array_id in chunk_data:
+                        consumer_array_id = array_id['consumerArrayID']  # consumerArrayID in all the chunk will be same
+                    pool_id = '%s:%s' % (sys_id, consumer_array_id)
+                    lsm_vol = Arcconf._arcconf_ld_to_lsm_vol(ld_info, pool_id, sys_id, cnt, consumer_array_id,
+                                                             str(ld_id), raid_level, ld_name)
```
Comment (cathay4t) [inline plugin/arcconf/arcconf.py]: The `ld_info` is already passed to `_arcconf_ld_to_lsm_vol()`. Please just read `raid_level`, `ld_id`, `ld_name` inside `_arcconf_ld_to_lsm_vol()`.

File: plugin/arcconf/arcconf.py [hunk 2]
```diff
@@ -439,9 +440,13 @@ def _arcconf_ld_to_lsm_vol(arcconf_ld,
...
         else:
             admin_status = Volume.ADMIN_STATE_ENABLED
-        plugin_data = "%s:%s:%s" % (ctrl_num, array_num, ld_num)
-
-        volume_id = array_num
+        # plugin_data = "%s:%s:%s:%s" % (ctrl_num, array_id, ld_id, raid_level)
+        stripe_size = arcconf_ld['StripeSize']
+        full_stripe_size = arcconf_ld['fullStripeSize']
+        ld_state = arcconf_ld['state']
+        plugin_data = {'ctrl_id': ctrl_num, 'array_id': array_id, 'ld_id': ld_id, 'ld_state': ld_state,
```
Comment (cathay4t) [inline plugin/arcconf/arcconf.py]: `plugin_data` should be `string`

File: plugin/arcconf/arcconf.py (hunk 1 above)
Comment (sharath-t) [reply plugin/arcconf/arcconf.py]: Done - commit id - a70a0a1a4fffd20be4239def1f0e0a2d91d1b41b

File: plugin/arcconf/arcconf.py (hunk 2 above)
Comment (sharath-t) [reply plugin/arcconf/arcconf.py]: Done - commit id - a70a0a1a4fffd20be4239def1f0e0a2d91d1b41b

**Comment by sharath-t** [comment]
Please accept commit id - a70a0a1a4fffd20be4239def1f0e0a2d91d1b41b and 8d2afdfdfca742c310045c3f6115bc36ae7809e8

**Comment by sharath-t** [comment]
reason for the multiple commits on this one is to make use of plugin_data for all volume dependent functions.
please accept a70a0a1a4fffd20be4239def1f0e0a2d91d1b41b, 8d2afdfdfca742c310045c3f6115bc36ae7809e8 and 094c51e088255fc19e33e052a9760c40d947100f commits

**Comment by sharath-t** [comment]
The last CI failed because of "volume_delete()" function. 
Due to the changes in "volume" functions "plugin_data", the older "volume_delete()" function will fail.
fix for this is already in place in PR #364

**Comment by tasleson** [comment]
Integrated into https://github.com/libstorage/libstoragemgmt/pull/386

**Pull request closed without merging**